### ISSUE TITLE
[BUG] Package fails to publish with OTP on `npm@10.8.0`

### ISSUE THREAD

**Issue by mikeseese** [npm/cli#7547]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When I run `npm publish`, after I successfully provide the OTP in the browser, the NPM CLI errors with:

```
400 Bad Request - PUT https://registry.npmjs.org/@incanta%2fconfig - child "otp" fails because ["otp" with value "[object Object]" fails to match the required pattern: /^\d+$/, "otp" length must be 64 characters long]
```

Downgrading to `npm@10.7.0` publishes the update without any issue.

### Expected Behavior

The package should publish without error.

### Steps To Reproduce

1. Use the latest NPM `npm i -g npm@10.8.0`
2. Clone `https://github.com/incanta/node-config`
3. Install yarn v1 `npm i -g yarn`
4. Install deps: `yarn`
5. Build TypeScript: `yarn build`
6. Change the `package.json` package name so you can publish it
7. Publish: `npm publish`
8. Provide OTP in browser
9. Return to CLI to see the version
10. See the error in the CLI

Repeat these steps after downgrading NPM `npm i -g npm@10.7.0` to see that you can publish.

### Environment

- npm: 10.8.0
- Node.js: 20.13.1
- OS Name: Windows 11 Pro
- System Model Name: Custom built desktop PC
- npm config:
```ini
; "user" config from C:\Users\seese\.npmrc

@fortawesome:registry = "https://npm.fontawesome.com/"
//npm.fontawesome.com/:_authToken = (protected)
//registry.npmjs.org/:_authToken = (protected)
msbuild_path = "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\Msbuild\\Current\\Bin\\MSBuild.exe"
msvs_version = "2022"

; node bin location = C:\Program Files\nodejs\node.exe
; node version = v20.13.1
; npm local prefix = D:\work\node-config
; npm version = 10.7.0
; cwd = D:\work\node-config
; HOME = C:\Users\seese
; Run `npm config ls -l` to show all defaults.
```

**Comment by fsegurai** [comment]
Same issue here, downgrading to `10.7.0` did the trick. No issues so far.

Issue is with **`10.8.0`**

**Comment by wraithgar** [comment]
This was fixed in https://github.com/npm/cli/pull/7545

**Comment by andrewangelle** [comment]
> This was fixed in #7545

No it wasnt

**Comment by mikeseese** [comment]
@andrewangelle I can confirm that `10.8.1` did fix this issue. Make sure you have the latest version of npm: `npm i -g npm` and check `npm --version` is `10.8.1` (latest) or higher. If you're still having issues, please open a new GH issue.
